consumer.go

```diff
@@ -523,6 +523,7 @@ func (child *partitionConsumer) parseRecords(batch *RecordBatch) ([]*ConsumerMes
 	var messages []*ConsumerMessage
 	var incomplete bool
 	prelude := true
+	originalOffset := child.offset
```

Comment:
should we save this value around L550 after we've already processed the messages?

Reply:
I don't think we should do that. If we are processing a non-compacted stream, the final message would set the offset correctly in the loop. In this case, the offset advancing check would fail incorrectly.